api/util/apps.go

```diff
@@ -66,6 +67,37 @@ func EnvVarsFromMap(env map[string]string) apps.EnvVars {
 	return vars
 }
 
+func UpdateEnvVars(oldEnvs []apps.EnvVar, newEnvs *map[string]string, toDelete *[]string) apps.EnvVars {
+	envMap := map[string]apps.EnvVar{}
+	for _, v := range oldEnvs {
+		envMap[v.Name] = v
+	}
+
+	if newEnvs != nil {
```

Comment:
I'd also just use a `map[string]string` here and no pointer to it.  You can just iterate over it if needed.

Reply:
here too